repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/CheckoutPreviousBuild.java

```diff
@@ -21,43 +19,29 @@ public CheckoutPreviousBuild(ProjectInspector inspector) {
     }
 
     protected void businessExecute() {
-        this.getLogger().debug("Testing previous build...");
+        this.getLogger().debug("Checking out previous build...");
 
         this.inspector.setPreviousBuildFlag(true);
 
-        Build previousBuild = BuildHelper.getLastBuildOfSameBranchOfStatusBeforeBuild(this.inspector.getBuild(), null);
-
-        if (previousBuild != null) {
-            this.inspector.setPreviousBuild(previousBuild);
-
-            this.getLogger().debug("Build: " + this.inspector.getBuild().getId());
-            this.getLogger().debug("Previous build: " + previousBuild.getId());
-
-            if (previousBuild.getBuildStatus() == BuildStatus.FAILED) {
-                this.shouldStop = false;
-
-                Git git;
-                try {
-                    git = Git.open(new File(inspector.getRepoLocalPath()));
-                    String commitCheckout = previousBuild.getCommit().getSha();
-                    commitCheckout = this.testCommitExistence(git, commitCheckout);
-                    if (commitCheckout != null) {
-                        this.getLogger().debug(
-                                "Get the commit " + commitCheckout + " for repo " + this.inspector.getRepoSlug());
-                        git.checkout().setName(commitCheckout).call();
-                    } else {
-                        this.addStepError("Error while getting the commit to checkout from the repo.");
-                        this.shouldStop = true;
-                        return;
-                    }
-                } catch (IOException | GitAPIException e) {
-                    e.printStackTrace();
-                }
+        Git git;
+        try {
+            git = Git.open(new File(inspector.getRepoLocalPath()));
+            String commitCheckout = inspector.getPreviousBuild().getCommit().getSha();
+            commitCheckout = this.testCommitExistence(git, commitCheckout);
+            if (commitCheckout != null) {
+                this.getLogger()
+                        .debug("Get the commit " + commitCheckout + " for repo " + this.inspector.getRepoSlug());
+                git.checkout().setName(commitCheckout).call();
+                this.state = ProjectState.PREVIOUSBUILDCHECKEDOUT;
             } else {
+                this.addStepError("Error while getting the commit to checkout from the repo.");
                 this.shouldStop = true;
+                this.state = ProjectState.PREVIOUSBUILDNOTCHECKEDOUT;
+                return;
             }
-        } else {
-            this.shouldStop = true;
+        } catch (IOException | GitAPIException e) {
+            e.printStackTrace();
```

Comment:
You should not use printStackTrace here, but set the shouldStop value and record error in the inspector using the addStepError method.